File: chef/Starters25/EQUALMEX.py
t=int(input())
for i in range(t):
    o=int(input())
    n=list(map(int,input().split(" ")))
    list1=[n[i] for i in range(o)]
    list2=[n[i] for i in range(o,2*o)]
    set1=set(list1)
    set2=set(list2)
    list1=list(set1)
    list2=list(set2)
    list1.sort()
    list2.sort()

    for k in range(o):
        if list1[0]!=0:
            MEX1=0
            break
        if k==0:
            MEX1=list1[0]
        else:
            present=list1[k]
            prev=list1[k-1]
            if (present-prev>1):
                MEX1=prev+1
                break
    for k in range(o):
        if list2[0]!=0:
            MEX2=0
            break
        if k==0:
            MEX2=list2[0]
        else:
            present=list2[k]
            prev=list2[k-1]
            if (present-prev>1):
                MEX2=prev+1
                break
    if MEX1==MEX2:
        print("YES")
    else:
        print("NO")

File: chef/Starters25/EQUALMEX.cpp
#include <iostream>
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int t, MEX1, MEX2;
    cin >> t;
    while (t--)
    {
        int n;
        cin >> n;
        set<int> s1;
        set<int> s2;
        for (int i = 0; i < n; i++)
        {
            int x;
            cin >> x;
            s1.insert(x);
        }
        for (int i = 0; i < n; i++)
        {
            int x;
            cin >> x;
            s2.insert(x);
        }
        sort(s1.begin(), s1.end());
        sort(s2.begin(), s2.end());
        set<int>::iterator it;
        int prev1, present1;
        for (it = s1.begin(); it != s1.end(); it++)
        {
            if (*it != 0)
            {
                MEX1 = 0;
                break;
            }
            if (it == s1.begin())
            {
                MEX1 = *it;
            }
            else
            {
                present1 = *it;
                prev1 = *(--it);
                if (present1 - prev1 > 1)
                {
                    MEX1 = prev1 + 1;
                    break;
                }
            }
        }
        int prev2, present2;
        for (it = s2.begin(); it != s2.end(); it++)
        {
            if (*it != 0)
            {
                MEX2 = 0;
                break;
            }
            if (it == s2.begin())
            {
                MEX2 = *it;
            }
            else
            {
                present2 = *it;
                prev2 = *(--it);
                if (present2 - prev2 > 1)
                {
                    MEX2 = prev2 + 1;
                    break;
                }
            }
        }
        if (MEX1 == MEX2)
        {
            cout << "YES" << endl;
        }
        else
        {
            cout << "NO" << endl;
        }
    }
}